From 54e8a0a03c998f82e1e40c5f4ed1bb446bd76c51 Mon Sep 17 00:00:00 2001
From: hgy59 <hpgy59@gmail.com>
Date: Fri, 6 Dec 2024 23:21:52 +0100
Subject: [PATCH] migrate requirements for crossenv from mk/crossenv to
 spk/python31*/crossenv

- use python version specific crossenv
- downgrade cryptography in default crossenv for python < 3.12 to v41.0.3
---
 mk/spksrc.crossenv.mk                         |  2 +-
 .../crossenv/requirements-default.txt         | 19 +++++++++++++++++++
 .../crossenv/requirements-frozenlist.txt      |  0
 .../crossenv/requirements-numpy-1.22.4.txt    |  0
 .../crossenv/requirements-numpy-1.24.4.txt    |  0
 .../crossenv/requirements-numpy-1.25.1.txt    |  0
 .../crossenv/requirements-pydantic_core.txt   |  0
 .../python310}/crossenv/requirements-yarl.txt |  0
 .../crossenv/requirements-default.txt         | 19 +++++++++++++++++++
 .../crossenv/requirements-frozenlist.txt      |  8 ++++++++
 .../crossenv/requirements-numpy-1.22.4.txt    | 14 ++++++++++++++
 .../crossenv/requirements-numpy-1.24.4.txt    | 14 ++++++++++++++
 .../crossenv/requirements-numpy-1.25.1.txt    | 14 ++++++++++++++
 .../crossenv/requirements-pydantic_core.txt   | 10 ++++++++++
 spk/python311/crossenv/requirements-yarl.txt  |  8 ++++++++
 .../crossenv/requirements-default.txt         |  0
 .../crossenv/requirements-frozenlist.txt      |  8 ++++++++
 .../crossenv/requirements-numpy-1.22.4.txt    | 14 ++++++++++++++
 .../crossenv/requirements-numpy-1.24.4.txt    | 14 ++++++++++++++
 .../crossenv/requirements-numpy-1.25.1.txt    | 14 ++++++++++++++
 .../crossenv/requirements-pydantic_core.txt   | 10 ++++++++++
 spk/python312/crossenv/requirements-yarl.txt  |  8 ++++++++
 .../crossenv/requirements-default.txt         | 19 +++++++++++++++++++
 .../crossenv/requirements-frozenlist.txt      |  8 ++++++++
 .../crossenv/requirements-numpy-1.22.4.txt    | 14 ++++++++++++++
 .../crossenv/requirements-numpy-1.24.4.txt    | 14 ++++++++++++++
 .../crossenv/requirements-numpy-1.25.1.txt    | 14 ++++++++++++++
 .../crossenv/requirements-pydantic_core.txt   | 10 ++++++++++
 spk/python313/crossenv/requirements-yarl.txt  |  8 ++++++++
 29 files changed, 262 insertions(+), 1 deletion(-)
 create mode 100644 spk/python310/crossenv/requirements-default.txt
 rename {mk => spk/python310}/crossenv/requirements-frozenlist.txt (100%)
 rename {mk => spk/python310}/crossenv/requirements-numpy-1.22.4.txt (100%)
 rename {mk => spk/python310}/crossenv/requirements-numpy-1.24.4.txt (100%)
 rename {mk => spk/python310}/crossenv/requirements-numpy-1.25.1.txt (100%)
 rename {mk => spk/python310}/crossenv/requirements-pydantic_core.txt (100%)
 rename {mk => spk/python310}/crossenv/requirements-yarl.txt (100%)
 create mode 100644 spk/python311/crossenv/requirements-default.txt
 create mode 100644 spk/python311/crossenv/requirements-frozenlist.txt
 create mode 100644 spk/python311/crossenv/requirements-numpy-1.22.4.txt
 create mode 100644 spk/python311/crossenv/requirements-numpy-1.24.4.txt
 create mode 100644 spk/python311/crossenv/requirements-numpy-1.25.1.txt
 create mode 100644 spk/python311/crossenv/requirements-pydantic_core.txt
 create mode 100644 spk/python311/crossenv/requirements-yarl.txt
 rename {mk => spk/python312}/crossenv/requirements-default.txt (100%)
 create mode 100644 spk/python312/crossenv/requirements-frozenlist.txt
 create mode 100644 spk/python312/crossenv/requirements-numpy-1.22.4.txt
 create mode 100644 spk/python312/crossenv/requirements-numpy-1.24.4.txt
 create mode 100644 spk/python312/crossenv/requirements-numpy-1.25.1.txt
 create mode 100644 spk/python312/crossenv/requirements-pydantic_core.txt
 create mode 100644 spk/python312/crossenv/requirements-yarl.txt
 create mode 100644 spk/python313/crossenv/requirements-default.txt
 create mode 100644 spk/python313/crossenv/requirements-frozenlist.txt
 create mode 100644 spk/python313/crossenv/requirements-numpy-1.22.4.txt
 create mode 100644 spk/python313/crossenv/requirements-numpy-1.24.4.txt
 create mode 100644 spk/python313/crossenv/requirements-numpy-1.25.1.txt
 create mode 100644 spk/python313/crossenv/requirements-pydantic_core.txt
 create mode 100644 spk/python313/crossenv/requirements-yarl.txt

diff --git mk/spksrc.crossenv.mk mk/spksrc.crossenv.mk
index bbcc4395..e0615607 100644
--- mk/spksrc.crossenv.mk
+++ mk/spksrc.crossenv.mk
@@ -33,7 +33,7 @@ PYTHON_LIB_NATIVE           = $(abspath $(PYTHON_WORK_DIR)/$(PYTHON_PKG_DIR)/bui
 PYTHON_LIB_CROSS            = $(abspath $(PYTHON_WORK_DIR)/$(PYTHON_PKG_DIR)/build/lib.linux-$(shell expr "$(TC_TARGET)" : '\([^-]*\)' )-$(PYTHON_PKG_VERS_MAJOR_MINOR))
 
 # wheel crossenv definitions
-CROSSENV_CONFIG_PATH = $(abspath $(WORK_DIR)/../../../mk/crossenv)
+CROSSENV_CONFIG_PATH = $(abspath $(WORK_DIR)/../../../spk/$(PYTHON_PKG_NAME)/crossenv)
 CROSSENV_CONFIG_DEFAULT = $(CROSSENV_CONFIG_PATH)/requirements-default.txt
 CROSSENV_PATH = $(abspath $(WORK_DIR)/crossenv-$(CROSSENV_BUILD_WHEEL)/)
 
diff --git spk/python310/crossenv/requirements-default.txt spk/python310/crossenv/requirements-default.txt
new file mode 100644
index 00000000..4b8f768e
--- /dev/null
+++ spk/python310/crossenv/requirements-default.txt
@@ -0,0 +1,19 @@
+pip==24.3.1
+pip-tools==7.4.1
+setuptools==75.4.0
+build==1.2.2
+wheel==0.45.0
+#
+cffi==1.17.1
+cryptography==41.0.3
+Cython==3.0.11
+flit==3.10.1
+maturin==1.7.4
+meson-python==0.17.1
+msgpack==1.0.5
+poetry==1.8.4
+scikit-build==0.18.1
+scikit-build-core==0.10.7
+setuptools-rust==1.10.2
+setuptools-scm==8.1.0
+typing_extensions==4.12.2
diff --git mk/crossenv/requirements-frozenlist.txt spk/python310/crossenv/requirements-frozenlist.txt
similarity index 100%
rename from mk/crossenv/requirements-frozenlist.txt
rename to spk/python310/crossenv/requirements-frozenlist.txt
diff --git mk/crossenv/requirements-numpy-1.22.4.txt spk/python310/crossenv/requirements-numpy-1.22.4.txt
similarity index 100%
rename from mk/crossenv/requirements-numpy-1.22.4.txt
rename to spk/python310/crossenv/requirements-numpy-1.22.4.txt
diff --git mk/crossenv/requirements-numpy-1.24.4.txt spk/python310/crossenv/requirements-numpy-1.24.4.txt
similarity index 100%
rename from mk/crossenv/requirements-numpy-1.24.4.txt
rename to spk/python310/crossenv/requirements-numpy-1.24.4.txt
diff --git mk/crossenv/requirements-numpy-1.25.1.txt spk/python310/crossenv/requirements-numpy-1.25.1.txt
similarity index 100%
rename from mk/crossenv/requirements-numpy-1.25.1.txt
rename to spk/python310/crossenv/requirements-numpy-1.25.1.txt
diff --git mk/crossenv/requirements-pydantic_core.txt spk/python310/crossenv/requirements-pydantic_core.txt
similarity index 100%
rename from mk/crossenv/requirements-pydantic_core.txt
rename to spk/python310/crossenv/requirements-pydantic_core.txt
diff --git mk/crossenv/requirements-yarl.txt spk/python310/crossenv/requirements-yarl.txt
similarity index 100%
rename from mk/crossenv/requirements-yarl.txt
rename to spk/python310/crossenv/requirements-yarl.txt
diff --git spk/python311/crossenv/requirements-default.txt spk/python311/crossenv/requirements-default.txt
new file mode 100644
index 00000000..4b8f768e
--- /dev/null
+++ spk/python311/crossenv/requirements-default.txt
@@ -0,0 +1,19 @@
+pip==24.3.1
+pip-tools==7.4.1
+setuptools==75.4.0
+build==1.2.2
+wheel==0.45.0
+#
+cffi==1.17.1
+cryptography==41.0.3
+Cython==3.0.11
+flit==3.10.1
+maturin==1.7.4
+meson-python==0.17.1
+msgpack==1.0.5
+poetry==1.8.4
+scikit-build==0.18.1
+scikit-build-core==0.10.7
+setuptools-rust==1.10.2
+setuptools-scm==8.1.0
+typing_extensions==4.12.2
diff --git spk/python311/crossenv/requirements-frozenlist.txt spk/python311/crossenv/requirements-frozenlist.txt
new file mode 100644
index 00000000..4edc3b10
--- /dev/null
+++ spk/python311/crossenv/requirements-frozenlist.txt
@@ -0,0 +1,8 @@
+pip==24.3.1
+pip-tools==7.4.1
+setuptools==75.4.0
+build==1.2.2
+wheel==0.45.0
+#
+Cython==3.0.11
+expandvars==0.12.0
diff --git spk/python311/crossenv/requirements-numpy-1.22.4.txt spk/python311/crossenv/requirements-numpy-1.22.4.txt
new file mode 100644
index 00000000..e740501c
--- /dev/null
+++ spk/python311/crossenv/requirements-numpy-1.22.4.txt
@@ -0,0 +1,14 @@
+pip==23.2.1
+setuptools==63.4.3
+wheel==0.41.2
+#
+cffi==1.15.1
+cryptography==41.0.3
+Cython==0.29.36
+flit==3.9.0
+maturin==1.2.3
+msgpack==1.0.5
+poetry==1.6.1
+scikit-build==0.17.6
+setuptools-rust==1.7.0
+setuptools-scm==7.1.0
diff --git spk/python311/crossenv/requirements-numpy-1.24.4.txt spk/python311/crossenv/requirements-numpy-1.24.4.txt
new file mode 100644
index 00000000..e740501c
--- /dev/null
+++ spk/python311/crossenv/requirements-numpy-1.24.4.txt
@@ -0,0 +1,14 @@
+pip==23.2.1
+setuptools==63.4.3
+wheel==0.41.2
+#
+cffi==1.15.1
+cryptography==41.0.3
+Cython==0.29.36
+flit==3.9.0
+maturin==1.2.3
+msgpack==1.0.5
+poetry==1.6.1
+scikit-build==0.17.6
+setuptools-rust==1.7.0
+setuptools-scm==7.1.0
diff --git spk/python311/crossenv/requirements-numpy-1.25.1.txt spk/python311/crossenv/requirements-numpy-1.25.1.txt
new file mode 100644
index 00000000..e740501c
--- /dev/null
+++ spk/python311/crossenv/requirements-numpy-1.25.1.txt
@@ -0,0 +1,14 @@
+pip==23.2.1
+setuptools==63.4.3
+wheel==0.41.2
+#
+cffi==1.15.1
+cryptography==41.0.3
+Cython==0.29.36
+flit==3.9.0
+maturin==1.2.3
+msgpack==1.0.5
+poetry==1.6.1
+scikit-build==0.17.6
+setuptools-rust==1.7.0
+setuptools-scm==7.1.0
diff --git spk/python311/crossenv/requirements-pydantic_core.txt spk/python311/crossenv/requirements-pydantic_core.txt
new file mode 100644
index 00000000..aab8e3a0
--- /dev/null
+++ spk/python311/crossenv/requirements-pydantic_core.txt
@@ -0,0 +1,10 @@
+pip==24.3.1
+pip-tools==7.4.1
+setuptools==75.4.0
+build==1.2.2
+wheel==0.45.0
+#
+# [pydantic_core] depends on Cython < 3.x
+Cython==0.29.37
+maturin==1.7.4
+typing_extensions==4.12.2
diff --git spk/python311/crossenv/requirements-yarl.txt spk/python311/crossenv/requirements-yarl.txt
new file mode 100644
index 00000000..4edc3b10
--- /dev/null
+++ spk/python311/crossenv/requirements-yarl.txt
@@ -0,0 +1,8 @@
+pip==24.3.1
+pip-tools==7.4.1
+setuptools==75.4.0
+build==1.2.2
+wheel==0.45.0
+#
+Cython==3.0.11
+expandvars==0.12.0
diff --git mk/crossenv/requirements-default.txt spk/python312/crossenv/requirements-default.txt
similarity index 100%
rename from mk/crossenv/requirements-default.txt
rename to spk/python312/crossenv/requirements-default.txt
diff --git spk/python312/crossenv/requirements-frozenlist.txt spk/python312/crossenv/requirements-frozenlist.txt
new file mode 100644
index 00000000..4edc3b10
--- /dev/null
+++ spk/python312/crossenv/requirements-frozenlist.txt
@@ -0,0 +1,8 @@
+pip==24.3.1
+pip-tools==7.4.1
+setuptools==75.4.0
+build==1.2.2
+wheel==0.45.0
+#
+Cython==3.0.11
+expandvars==0.12.0
diff --git spk/python312/crossenv/requirements-numpy-1.22.4.txt spk/python312/crossenv/requirements-numpy-1.22.4.txt
new file mode 100644
index 00000000..e740501c
--- /dev/null
+++ spk/python312/crossenv/requirements-numpy-1.22.4.txt
@@ -0,0 +1,14 @@
+pip==23.2.1
+setuptools==63.4.3
+wheel==0.41.2
+#
+cffi==1.15.1
+cryptography==41.0.3
+Cython==0.29.36
+flit==3.9.0
+maturin==1.2.3
+msgpack==1.0.5
+poetry==1.6.1
+scikit-build==0.17.6
+setuptools-rust==1.7.0
+setuptools-scm==7.1.0
diff --git spk/python312/crossenv/requirements-numpy-1.24.4.txt spk/python312/crossenv/requirements-numpy-1.24.4.txt
new file mode 100644
index 00000000..e740501c
--- /dev/null
+++ spk/python312/crossenv/requirements-numpy-1.24.4.txt
@@ -0,0 +1,14 @@
+pip==23.2.1
+setuptools==63.4.3
+wheel==0.41.2
+#
+cffi==1.15.1
+cryptography==41.0.3
+Cython==0.29.36
+flit==3.9.0
+maturin==1.2.3
+msgpack==1.0.5
+poetry==1.6.1
+scikit-build==0.17.6
+setuptools-rust==1.7.0
+setuptools-scm==7.1.0
diff --git spk/python312/crossenv/requirements-numpy-1.25.1.txt spk/python312/crossenv/requirements-numpy-1.25.1.txt
new file mode 100644
index 00000000..e740501c
--- /dev/null
+++ spk/python312/crossenv/requirements-numpy-1.25.1.txt
@@ -0,0 +1,14 @@
+pip==23.2.1
+setuptools==63.4.3
+wheel==0.41.2
+#
+cffi==1.15.1
+cryptography==41.0.3
+Cython==0.29.36
+flit==3.9.0
+maturin==1.2.3
+msgpack==1.0.5
+poetry==1.6.1
+scikit-build==0.17.6
+setuptools-rust==1.7.0
+setuptools-scm==7.1.0
diff --git spk/python312/crossenv/requirements-pydantic_core.txt spk/python312/crossenv/requirements-pydantic_core.txt
new file mode 100644
index 00000000..aab8e3a0
--- /dev/null
+++ spk/python312/crossenv/requirements-pydantic_core.txt
@@ -0,0 +1,10 @@
+pip==24.3.1
+pip-tools==7.4.1
+setuptools==75.4.0
+build==1.2.2
+wheel==0.45.0
+#
+# [pydantic_core] depends on Cython < 3.x
+Cython==0.29.37
+maturin==1.7.4
+typing_extensions==4.12.2
diff --git spk/python312/crossenv/requirements-yarl.txt spk/python312/crossenv/requirements-yarl.txt
new file mode 100644
index 00000000..4edc3b10
--- /dev/null
+++ spk/python312/crossenv/requirements-yarl.txt
@@ -0,0 +1,8 @@
+pip==24.3.1
+pip-tools==7.4.1
+setuptools==75.4.0
+build==1.2.2
+wheel==0.45.0
+#
+Cython==3.0.11
+expandvars==0.12.0
diff --git spk/python313/crossenv/requirements-default.txt spk/python313/crossenv/requirements-default.txt
new file mode 100644
index 00000000..9400a967
--- /dev/null
+++ spk/python313/crossenv/requirements-default.txt
@@ -0,0 +1,19 @@
+pip==24.3.1
+pip-tools==7.4.1
+setuptools==75.4.0
+build==1.2.2
+wheel==0.45.0
+#
+cffi==1.17.1
+cryptography==43.0.3
+Cython==3.0.11
+flit==3.10.1
+maturin==1.7.4
+meson-python==0.17.1
+msgpack==1.0.5
+poetry==1.8.4
+scikit-build==0.18.1
+scikit-build-core==0.10.7
+setuptools-rust==1.10.2
+setuptools-scm==8.1.0
+typing_extensions==4.12.2
diff --git spk/python313/crossenv/requirements-frozenlist.txt spk/python313/crossenv/requirements-frozenlist.txt
new file mode 100644
index 00000000..4edc3b10
--- /dev/null
+++ spk/python313/crossenv/requirements-frozenlist.txt
@@ -0,0 +1,8 @@
+pip==24.3.1
+pip-tools==7.4.1
+setuptools==75.4.0
+build==1.2.2
+wheel==0.45.0
+#
+Cython==3.0.11
+expandvars==0.12.0
diff --git spk/python313/crossenv/requirements-numpy-1.22.4.txt spk/python313/crossenv/requirements-numpy-1.22.4.txt
new file mode 100644
index 00000000..e740501c
--- /dev/null
+++ spk/python313/crossenv/requirements-numpy-1.22.4.txt
@@ -0,0 +1,14 @@
+pip==23.2.1
+setuptools==63.4.3
+wheel==0.41.2
+#
+cffi==1.15.1
+cryptography==41.0.3
+Cython==0.29.36
+flit==3.9.0
+maturin==1.2.3
+msgpack==1.0.5
+poetry==1.6.1
+scikit-build==0.17.6
+setuptools-rust==1.7.0
+setuptools-scm==7.1.0
diff --git spk/python313/crossenv/requirements-numpy-1.24.4.txt spk/python313/crossenv/requirements-numpy-1.24.4.txt
new file mode 100644
index 00000000..e740501c
--- /dev/null
+++ spk/python313/crossenv/requirements-numpy-1.24.4.txt
@@ -0,0 +1,14 @@
+pip==23.2.1
+setuptools==63.4.3
+wheel==0.41.2
+#
+cffi==1.15.1
+cryptography==41.0.3
+Cython==0.29.36
+flit==3.9.0
+maturin==1.2.3
+msgpack==1.0.5
+poetry==1.6.1
+scikit-build==0.17.6
+setuptools-rust==1.7.0
+setuptools-scm==7.1.0
diff --git spk/python313/crossenv/requirements-numpy-1.25.1.txt spk/python313/crossenv/requirements-numpy-1.25.1.txt
new file mode 100644
index 00000000..e740501c
--- /dev/null
+++ spk/python313/crossenv/requirements-numpy-1.25.1.txt
@@ -0,0 +1,14 @@
+pip==23.2.1
+setuptools==63.4.3
+wheel==0.41.2
+#
+cffi==1.15.1
+cryptography==41.0.3
+Cython==0.29.36
+flit==3.9.0
+maturin==1.2.3
+msgpack==1.0.5
+poetry==1.6.1
+scikit-build==0.17.6
+setuptools-rust==1.7.0
+setuptools-scm==7.1.0
diff --git spk/python313/crossenv/requirements-pydantic_core.txt spk/python313/crossenv/requirements-pydantic_core.txt
new file mode 100644
index 00000000..aab8e3a0
--- /dev/null
+++ spk/python313/crossenv/requirements-pydantic_core.txt
@@ -0,0 +1,10 @@
+pip==24.3.1
+pip-tools==7.4.1
+setuptools==75.4.0
+build==1.2.2
+wheel==0.45.0
+#
+# [pydantic_core] depends on Cython < 3.x
+Cython==0.29.37
+maturin==1.7.4
+typing_extensions==4.12.2
diff --git spk/python313/crossenv/requirements-yarl.txt spk/python313/crossenv/requirements-yarl.txt
new file mode 100644
index 00000000..4edc3b10
--- /dev/null
+++ spk/python313/crossenv/requirements-yarl.txt
@@ -0,0 +1,8 @@
+pip==24.3.1
+pip-tools==7.4.1
+setuptools==75.4.0
+build==1.2.2
+wheel==0.45.0
+#
+Cython==3.0.11
+expandvars==0.12.0
-- 
2.39.2

